goad.sh

```diff
@@ -486,27 +520,32 @@ stop(){
       fi
       ;;
     "azure")
-      az vm stop --ids $(az vm list --resource-group $LAB --query "[].id" -o tsv)
+      # shellcheck disable=SC2046
+      az vm stop --ids $(az vm list --resource-group "$LAB" --query "[].id" -o tsv)
       status
       ;;
+    *)
+          echo "unknown provider"
+      ;;
   esac
 }
 
 restart(){
   case $PROVIDER in
-    "virtualbox"|"vmware")
-          cd "ad/$LAB/providers/$PROVIDER"
+    "virtualbox"|"vmware"|"vmware_esxi")
+        (
+          cd "ad/$LAB/providers/$PROVIDER" || { echo "provider dir does not exist"; exit 1; }
           echo "${OK} restart start vms"
           vagrant reload
-          cd -
```

Comment:
why removing the comeback to previous location ?

Reply:
This was done referring to the https://www.shellcheck.net/wiki/SC2103 suggestion, using subshell scope to not have to `cd` back.